libs/utils/android/benchmark.py

```diff
@@ -0,0 +1,287 @@
+
+        # Bootup settings
+        parser.add_argument('--boot-image', type=str,
+                default=None,
+                help='Path of the Android boot.img to be used')
+        parser.add_argument('--boot-timeout', type=int,
+                default=20,
+                help='Timeout in [s] to wait after a reboot (default 20)')
+
+        # Android settings
+        parser.add_argument('--android-device', type=str,
+                default=None,
+                help='Identifier of the Android target to use')
+        parser.add_argument('--android-home', type=str,
+                default=None,
+                help='Path used to configure ANDROID_HOME')
+
+        # Test customization
+        parser.add_argument('--results-dir', type=str,
+                default=None,
+                help='Results folder, '
+                     'if specified override test defaults')
+        parser.add_argument('--collect', type=str,
+                default=None,
+                help='Set of metrics to collect, '
+                     'e.g. "energy systrace_30" to sample energy and collect a 30s systrace, '
+                     'if specified overrides test defaults')
+
+        # Measurements settings
+        parser.add_argument('--iio-channel-map', type=str,
+                default=None,
+                help='List of IIO channels to sample, '
+                     'e.g. "ch0:0,ch3:1" to sample CHs 0 and 3, '
+                     'if specified overrides test defaults')
+
+        # Parse command line arguments
+        return parser.parse_args()
+
+
+    def _getBmConf(self):
+        if self.bm_conf is None:
+            msg = 'Benchmark subclasses must override the `bm_conf` attribute'
+            raise NotImplementedError(msg)
+
+        # Override default configuration with command line parameters
+        if self.args.android_device:
+            self.bm_conf['device'] = self.args.android_device
+        if self.args.android_home:
+            self.bm_conf['ANDROID_HOME'] = self.args.android_home
+        if self.args.results_dir:
+            self.bm_conf['results_dir'] = self.args.results_dir
+        if self.args.collect:
+            self.bm_collect = self.args.collect
+
+        # Override energy meter configuration
+        if self.args.iio_channel_map:
+            em = {
+                'instrument'  : 'acme',
+                'channel_map' : {},
+            }
+            for ch in self.args.iio_channel_map.split(','):
+                ch_name, ch_id = ch.split(':')
+                em['channel_map'][ch_name] = ch_id
+            self.bm_conf['emeter'] = em
+            self._log.info('Using ACME energy meter channels: %s', em)
+
+        # Override EM if energy collection not required
+        if 'energy' not in self.bm_collect:
+            try:
+                self.bm_conf.pop('emeter')
+            except:
+                pass
+
+        return self.bm_conf
+
+    def _getBmName(self):
```

Comment:
Right, that was to keep an attribute getter in sync with a attribute name... well... forget it! ;-)
You right... it's a private method ad name clashing should not be an issue here.